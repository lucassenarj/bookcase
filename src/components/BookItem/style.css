.single-blog {
	background: #ffffff;
}

.single-blog .blog-image a {
	display: block;
	cursor: pointer;
}

.single-blog .blog-image a img {
	width: 100%;
	max-height: 380px;
}

.single-blog .blog-content {
	padding: 20px 10px;
}

.single-blog .blog-content .blog-title a {
	font-size: 24px;
	font-weight: 600;
	color: #222;
	-webkit-transition: all 0.3s ease-out 0s;
	-moz-transition: all 0.3s ease-out 0s;
	-ms-transition: all 0.3s ease-out 0s;
	-o-transition: all 0.3s ease-out 0s;
	transition: all 0.3s ease-out 0s;
	cursor: pointer;
}

@media only screen and (min-width: 992px) and (max-width: 1199px) {
	.single-blog .blog-content .blog-title a {
		font-size: 20px;
	}
}

@media (max-width: 767px) {
	.single-blog .blog-content .blog-title a {
		font-size: 20px;
	}
}

.single-blog .blog-content .blog-title a:hover {
	color: #f14836;
}

.single-blog .blog-content .blog-author {
	padding-top: 20px;
}

.single-blog .blog-content .blog-author .author-image img {
	width: 65px;
	border-radius: 50%;
}

.single-blog .blog-content .blog-author .author-content {
	padding-left: 20px;
}

.single-blog .blog-content .blog-author .author-content .sub-title {
	font-size: 14px;
	color: #f14836;
	font-weight: 400;
}

.single-blog .blog-content .blog-author .author-content .text {
	font-size: 16px;
	font-weight: 600;
	margin-top: 5px;
}

.btn-click {
	cursor: pointer;
}