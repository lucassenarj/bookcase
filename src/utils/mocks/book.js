const book = {
  "kind": "books#volume",
  "id": "DKcWE3WXoj8C",
  "etag": "yCPnghuxncA",
  "selfLink": "https://content-books.googleapis.com/books/v1/volumes/DKcWE3WXoj8C",
  "volumeInfo": {
    "title": "Harry Potter and International Relations",
    "authors": [
      "Daniel H. Nexon",
      "Iver B. Neumann"
    ],
    "publisher": "Rowman & Littlefield",
    "publishedDate": "2006",
    "description": "Why not take seriously the claim that Harry Potter's world intertwines with our own? In this timely yet otherworldly volume, more than a dozen scholars of international relations join hands to demonstrate how this well-loved artifact of popular culture reflects and shapes our own lifeworld. A wide range of historical and sociological sources shows how Harry's world contains aspects of our own. Practices such as quidditch dovetail quite clearly with muggle sports, and the very British-ness of the books has, in translation into languages such as Turkish and Arabic, been transformed to reflect these unique cultures. Chapters on the political economy of the franchise as well as the scholarly problems of studying popular culture frame what is essentially a highly info-taining read. A landmark in muggle studies The Daily Prophet.",
    "industryIdentifiers": [
      {
        "type": "ISBN_10",
        "identifier": "0742539598"
      },
      {
        "type": "ISBN_13",
        "identifier": "9780742539594"
      }
    ],
    "readingModes": {
      "text": false,
      "image": true
    },
    "pageCount": 245,
    "printedPageCount": 262,
    "dimensions": {
      "height": "24.00 cm",
      "width": "16.30 cm",
      "thickness": "1.80 cm"
    },
    "printType": "BOOK",
    "categories": [
      "Literary Criticism / Science Fiction & Fantasy",
      "Literary Criticism / Children's & Young Adult Literature",
      "Political Science / General"
    ],
    "averageRating": 3.5,
    "ratingsCount": 8,
    "maturityRating": "NOT_MATURE",
    "allowAnonLogging": false,
    "contentVersion": "preview-1.0.0",
    "panelizationSummary": {
      "containsEpubBubbles": false,
      "containsImageBubbles": false
    },
    "imageLinks": {
      "smallThumbnail": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE71XDIwUM_yy7JuWypUulg3Voa-SMlqdcbnDPZkTP8XjV6ARgP1uovLfWs50i9HIzxHynC1uNHJvil7-6rBQIuBYdh7RGrKjX7PLbXmKGEexaPUJtCo&source=gbs_api",
      "thumbnail": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE72B5UJgHNPjBlNhjuEpojz1haNfRzQSBz4lIbth_v3rkFm0YARlNEiDDxI-xSreA2QGDKU39PfbCeY8skvSa-IAimVoqHejyu9o31_V6N3wjHLVYxw&source=gbs_api",
      "small": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=2&edge=curl&imgtk=AFLRE72_-CmGE480vRFL6yvWXyOHk1RbsohuvSq1uHKxsr4fmFHFU2YRCXpJXpK0dFgSlfRd3A9nzIW4FyvO0l_890CH08RplY2F2ZxdPC1uILMcKmbZJys&source=gbs_api",
      "medium": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=3&edge=curl&imgtk=AFLRE73ESDAsqMXXVa91zIt6-P5M2nG9trgPHygwPcI6v-CqUlE0aCh0DCNBU6Nkg3YxxuIYz_Lf6n85LYsU0dUQnkHRkQT095Cq2PWowMtvzDp_3fmHHBs&source=gbs_api",
      "large": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=4&edge=curl&imgtk=AFLRE72-vzgPQMA8BlmCT_e3XSo8Hw_iBCYWazJLXnt2H5P3rr0M2Ft5SFX3FXz_ZDJ8dJprmvTMd0BuX2jifJ-91F9lz6-9j1sN3IxBxi3RsyHcOi-8QMo&source=gbs_api",
      "extraLarge": "http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=6&edge=curl&imgtk=AFLRE739nnAXkWVnBbemsAEGU-6rIAcaKKji4MzvhvgapGw2A_VHooozCVw3KiOabNTJoA-87lR_zUtw5w2r_2sLRXDXhMOSKyys7UgKLqq2VwQR1c1B6bU&source=gbs_api"
    },
    "language": "en",
    "previewLink": "http://books.google.pt/books?id=DKcWE3WXoj8C&hl=&source=gbs_api",
    "infoLink": "https://play.google.com/store/books/details?id=DKcWE3WXoj8C&source=gbs_api",
    "canonicalVolumeLink": "https://play.google.com/store/books/details?id=DKcWE3WXoj8C"
  },
  "saleInfo": {
    "country": "PT",
    "saleability": "NOT_FOR_SALE",
    "isEbook": false
  },
  "accessInfo": {
    "country": "PT",
    "viewability": "PARTIAL",
    "embeddable": true,
    "publicDomain": false,
    "textToSpeechPermission": "ALLOWED",
    "epub": {
      "isAvailable": false
    },
    "pdf": {
      "isAvailable": true,
      "acsTokenLink": "http://books.google.pt/books/download/Harry_Potter_and_International_Relations-sample-pdf.acsm?id=DKcWE3WXoj8C&format=pdf&output=acs4_fulfillment_token&dl_type=sample&source=gbs_api"
    },
    "webReaderLink": "http://play.google.com/books/reader?id=DKcWE3WXoj8C&hl=&printsec=frontcover&source=gbs_api",
    "accessViewStatus": "SAMPLE",
    "quoteSharingAllowed": false
  }
};

export default book;
