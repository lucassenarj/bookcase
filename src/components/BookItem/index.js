import React from "react";
import PropTypes from "prop-types";
import ImageAuthor from "./../../assets/images/author-1.jpg";
import { useHistory } from "react-router-dom";
import { generateSlug } from "./../../utils/helpers";
import "./style.css";

function BookItem({ book }) {
  const slug = generateSlug(book.title);
  const history = useHistory();

  function handleClick() {
    const location = {
      pathname: `/details/${slug}`,
      state: { book }
    };
    history.push(location);
  }
  return (
    <div className="col-lg-4 col-md-6 col-sm-8 btn-click" onClick={() => handleClick()} data-testid="btn-click">
      <div className="single-blog mt-30 wow fadeInUpBig" data-wow-duration="1s" data-wow-delay="0.4s">
        <div className="blog-image">
          <a>
            <img src={book.imageLinks.thumbnail} alt="news" />
          </a>
        </div>
        <div className="blog-content">
          <h4 className="blog-title">
            <a>{ book.title }</a>
          </h4>
          <div className="blog-author d-flex align-items-center">
            <div className="author-image">
              <img src={ImageAuthor} alt="author" />
            </div>
            <div className="author-content media-body">
              <h6 className="sub-title">{ book.authors[0] }</h6>
              <p className="text">{ book.publishedDate }</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

BookItem.propTypes = {
  book: PropTypes.shape({
    title: PropTypes.string.isRequired,
    imageLinks: PropTypes.object.isRequired,
    authors: PropTypes.arrayOf(PropTypes.string).isRequired,
    publishedDate: PropTypes.string,
    averageRating: PropTypes.number,
  }).isRequired,
}

export default BookItem;
