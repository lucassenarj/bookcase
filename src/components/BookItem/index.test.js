import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";
import { generateSlug } from "./../../utils/helpers";
import BookItem from "./";
import book from "./../../utils/mocks/book";

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("BookItem Component", () => {
  it("renders correctly", () => {
    const { volumeInfo } = book;
    const { container } = render(<BookItem book={volumeInfo} />);

    expect(container.firstChild).toMatchSnapshot();
  });
  
  it("history push to details page", () => {
    const { volumeInfo } = book;
    const slug = generateSlug(volumeInfo.title);
    const location = {
      pathname: `/details/${slug}`,
      state: { book: volumeInfo }
    };

    const { getByTestId } = render(
      <MemoryRouter>
        <BookItem book={volumeInfo} />
      </MemoryRouter>
    );
  
    fireEvent.click(getByTestId("btn-click"));
  
    expect(mockHistoryPush).toHaveBeenCalledWith(location);
  });
});