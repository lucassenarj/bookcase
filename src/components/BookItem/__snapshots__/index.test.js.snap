// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem Component renders correctly 1`] = `
<div
  class="col-lg-4 col-md-6 col-sm-8 btn-click"
  data-testid="btn-click"
>
  <div
    class="single-blog mt-30 wow fadeInUpBig"
    data-wow-delay="0.4s"
    data-wow-duration="1s"
  >
    <div
      class="blog-image"
    >
      <a>
        <img
          alt="news"
          src="http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE72B5UJgHNPjBlNhjuEpojz1haNfRzQSBz4lIbth_v3rkFm0YARlNEiDDxI-xSreA2QGDKU39PfbCeY8skvSa-IAimVoqHejyu9o31_V6N3wjHLVYxw&source=gbs_api"
        />
      </a>
    </div>
    <div
      class="blog-content"
    >
      <h4
        class="blog-title"
      >
        <a>
          Harry Potter and International Relations
        </a>
      </h4>
      <div
        class="blog-author d-flex align-items-center"
      >
        <div
          class="author-image"
        >
          <img
            alt="author"
            src="author-1.jpg"
          />
        </div>
        <div
          class="author-content media-body"
        >
          <h6
            class="sub-title"
          >
            Daniel H. Nexon
          </h6>
          <p
            class="text"
          >
            2006
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
